servicelayer/taskqueue.py

```diff
@@ -275,9 +248,8 @@ def checkout_task(self, task_id, stage):
 
         pipe.srem(self.pending_key, task_id)
         pipe.sadd(self.running_key, task_id)
-        pipe.set(self.start_key, pack_now())
+        pipe.set(self.start_key, pack_now(), nx=True)
```

Comment:
I’m not entirely sure setting the "start_time" timestamp here is necessary in the first place as a worker shouldn’t be able to check out a task without `add_task` having been executed before. (The same probably also applies for `pipe.sadd(self.key, self.name)`.)

Didn’t change it because the primary goal of this PR is to fix the timestamps and not to refactor, but I’d still be interested to understand if there’s a legit case where this is relevant or if it’s merely as a precaution?